i1006: add "pointscoring" sample contest (v.1)

clevengr opened the pull request:
### Description of what the PR does
Adds to the PC2 `samps/contests` folder a new contest named "pointscoring".  The initial version of this contest has three (identical) problems in it, all based on the PC2 samps "Sumit" problem.  Currently, "problem a" in the sample contest is configured to use only Problem Package Specification defaults for scoring -- including that it does not contain any sub-folders (other than just `sample` and `secret`) beneath the `data` folder.  

Problems 'b' and 'c' are currently just duplicates of problem 'a'; the intent is eventually to update those problems (and add more if necessary) to use a variety of other "point scoring" configuration settings/flags.

### Issue which the PR addresses
Partial work towards #1006.

### Environment in which the PR was developed (OS,IDE, Java version, etc.)
Windows 10, Eclipse Version: 2020-12 (4.18.0), Java version "1.8.0_271"

### Precise steps for _testing_ the PR
- start a PC2 Server using `--load pointscoring`
- start a PC2 Admin.
- Verify that the loaded contest is configured with three problems, named "Sumit a", "Sumit b", and "Sumit c".
- Edit the "Sumit a" problem; examine the "Test Data Files" tab and ensure that there are three data file sets -- one from "sample" and two from "secret".

clevengr committed:
i1006: add "pointscoring" sample contest (v.1)

johnbrvc commented on samps/contests/pointscoring/config/contest.yaml:
```diff
@@ -0,0 +1,112 @@
+
+name: Point Scoring Contest  
+short-name: pointscoring
+
```
`contest.yaml `should have a "`scoreboard_type:`" property, either "`pass-fail`" or "`score`".  This is how the CCS knows how to score problems.  [Reference is spec for contests](https://ccs-specs.icpc.io/2023-06/contest_api#contests)  Note, however, if you set it to "score" at the moment, you will receive an Invalid configuration saying that "score" is Invalid only "pass-fail" is allowed (`ContestSnakeYamlLoader.java`, around line 616.  Maybe just add it as `pass-fail` now, with a comment saying we have to change it to score once `ContestSnakeYamlLoader` is fixed.

clevengr replied:
Fixed in latest push.  Set scoreboard_type to "pass-fail" in `contest.yaml`, with a comment that it needs to be changed to "score" once ContestSnakeYAMLLoader is updated.

johnbrvc commented on samps/contests/pointscoring/config/a/problem.yaml:
```diff
@@ -0,0 +1,12 @@
+--- 
+
+name: a
+type: scoring
```
While it is harmless to have the `type: scoring` in the `problem.yaml`, and, it part of the program package specification, it is of no use for a CCS since the `contest.yaml `file specifies if the contest is "score" or "pass-fail" using the `scoreboard_type` property.  Tou can't have a mixed contest (pass-fail and score).  If the `contest.yaml` says it is **score**, then all problems are assumed to be **score**.

clevengr replied:
I guess there is a "slight" justification for having `type: scoring` in a `problem.yaml` file:  problems get put in archives separate from the contest that originally used them; if someone later pulls the problem out of an archive then it's useful to have a flag that makes it clear that "this problem is expected to use point scoring".    (That doesn't mean it's particularly useful in a PC2 sample contest, but I think I'll leave it there just for completeness...)

johnbrvc reviewed (changes requested):
Other than adding the scoreboard_type: property to contest.yaml, and noting the meaningless type: in the problem.yaml, this looks fine.
Once the scoreboard_type is added, I can approve this.

clevengr committed:
i1006: update "pointscoring" contest.yaml to specify "scoreboard-type"

Note:  currently the "scoreboard_type" is set to "pass-fail", because the ContestSnakeYAMLLoader does not recognize "scoring" as a legitimate value.  This needs to be updated once ContestSnakeYAMLLoader is updated to recognize "scoring".
